### PR TITLE
upgrade riemann-client to 0.2.6

### LINKED ISSUE #25: reconnection after a riemann restart can take a long time (minutes)
It was observed that after a riemann restart, Logstash would not log errors or produce data to riemann for up to 16 minutes (observed by a user). After some time a host unreachable exception is raised and the connection is re-established.

Thread dump during that pause can show the plugin stuck on writing to the socket:

<details>
  <summary>Thread dump here</summary>
  
```

"[riemann]>worker0" #41 daemon prio=5 os_prio=0 cpu=42895.26ms elapsed=3200.92s tid=0x00007f56b404b000 nid=0x40 runnable  [0x00007f56f5eb8000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.FileDispatcherImpl.write0(java.base@11.0.6/Native Method)
	at sun.nio.ch.SocketDispatcher.write(java.base@11.0.6/SocketDispatcher.java:47)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(java.base@11.0.6/IOUtil.java:113)
	at sun.nio.ch.IOUtil.write(java.base@11.0.6/IOUtil.java:79)
	at sun.nio.ch.IOUtil.write(java.base@11.0.6/IOUtil.java:50)
	at sun.nio.ch.SocketChannelImpl.write(java.base@11.0.6/SocketChannelImpl.java:463)
	at org.jruby.util.io.PosixShim.write(PosixShim.java:106)
	at org.jruby.util.io.OpenFile$3.run(OpenFile.java:1347)
	at org.jruby.util.io.OpenFile$3.run(OpenFile.java:1338)
	at org.jruby.RubyThread.executeTask(RubyThread.java:1586)
	at org.jruby.RubyThread.executeTask(RubyThread.java:1572)
	at org.jruby.util.io.OpenFile.writeInternal(OpenFile.java:2268)
	at org.jruby.util.io.OpenFile.binwrite(OpenFile.java:2209)
	at org.jruby.util.io.OpenFile.fwrite(OpenFile.java:2076)
	at org.jruby.RubyIO.write(RubyIO.java:1476)
	at org.jruby.RubyIO.write(RubyIO.java:1432)
	at org.jruby.RubyIO$INVOKER$i$write.call(RubyIO$INVOKER$i$write.gen)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:172)
	at org.jruby.RubyIO.op_append(RubyIO.java:1491)
	at java.lang.invoke.LambdaForm$DMH/0x00000008406b5c40.invokeVirtual(java.base@11.0.6/LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x00000008406d3440.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.DelegatingMethodHandle$Holder.delegate(java.base@11.0.6/DelegatingMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000840689440.guard(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.DelegatingMethodHandle$Holder.delegate(java.base@11.0.6/DelegatingMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000840689440.guard(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.linkToCallSite(java.base@11.0.6/Invokers$Holder)
	at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.riemann_minus_client_minus_0_dot_2_dot_1.lib.riemann.client.tcp.RUBY$block$send_recv$1(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/riemann-client-0.2.1/lib/riemann/client/tcp.rb:51)
	at java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.base@11.0.6/DirectMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000841370440.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.invokeExact_MT(java.base@11.0.6/Invokers$Holder)
	at org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:146)
	at org.jruby.runtime.BlockBody.yield(BlockBody.java:114)
	at org.jruby.runtime.Block.yield(Block.java:170)
	at org.jruby.ir.runtime.IRRuntimeHelpers.yield(IRRuntimeHelpers.java:499)
	at org.jruby.ir.targets.YieldSite.yield(YieldSite.java:105)
	at java.lang.invoke.LambdaForm$DMH/0x00000008406de840.invokeVirtual(java.base@11.0.6/LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x0000000840700840.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.linkToCallSite(java.base@11.0.6/Invokers$Holder)
	at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.riemann_minus_client_minus_0_dot_2_dot_1.lib.riemann.client.tcp.RUBY$block$with_connection$1(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/riemann-client-0.2.1/lib/riemann/client/tcp.rb:65)
	at java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.base@11.0.6/DirectMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000841370040.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.invokeExact_MT(java.base@11.0.6/Invokers$Holder)
	at org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:146)
	at org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:85)
	at org.jruby.runtime.Block.yieldSpecific(Block.java:139)
	at org.jruby.ext.thread.Mutex.synchronize(Mutex.java:164)
	at java.lang.invoke.LambdaForm$DMH/0x00000008406b5c40.invokeVirtual(java.base@11.0.6/LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x00000008406d3440.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x00000008406da440.guardWithCatch(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x00000008406db040.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.DelegatingMethodHandle$Holder.delegate(java.base@11.0.6/DelegatingMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000840689440.guard(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.DelegatingMethodHandle$Holder.delegate(java.base@11.0.6/DelegatingMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000840689440.guard(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.linkToCallSite(java.base@11.0.6/Invokers$Holder)
	at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.riemann_minus_client_minus_0_dot_2_dot_1.lib.riemann.client.tcp.RUBY$method$with_connection$0(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/riemann-client-0.2.1/lib/riemann/client/tcp.rb:62)
	at java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.base@11.0.6/DirectMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x00000008408b4440.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x00000008406da440.guardWithCatch(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x00000008406db040.invoke(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.DelegatingMethodHandle$Holder.delegate(java.base@11.0.6/DelegatingMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000840689440.guard(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.DelegatingMethodHandle$Holder.delegate(java.base@11.0.6/DelegatingMethodHandle$Holder)
	at java.lang.invoke.LambdaForm$MH/0x0000000840689440.guard(java.base@11.0.6/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.linkToCallSite(java.base@11.0.6/Invokers$Holder)
	at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.riemann_minus_client_minus_0_dot_2_dot_1.lib.riemann.client.tcp.RUBY$method$send_recv$0(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/riemann-client-0.2.1/lib/riemann/client/tcp.rb:50)
```

</details>

### PR DESCRIPTION
**Pull request by colinsurprenant**
Fixes #25 
upgrade riemann-client to 0.2.6 to benefit timeout support
improved a spec
bumped versions to 3.0.7

### CODE REVIEW

**Comment by colinsurprenant** [comment]
For some reason the last commit does not show up here yet 🤔

**Review by roaksoax — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 3.0.7
+  - Updated riemann-client gem to 0.2.6 to benefit timeout support
```
Comment (roaksoax) [inline CHANGELOG.md]: Minor nit: should we link the issue it fixes here? (as how we are linking other changelog entries?)

Comment (colinsurprenant) [reply CHANGELOG.md]: I did but for some reason github has not seen that last commit yet. looks like github is having some issues at the moment. Will wait a bit.

**Review by jsvd — APPROVED** [review]
LGTM

**Comment by flaper87** [comment]
Does the newer version of the riemann client library use the `timeout` internally? or is that something we should add here?

**Comment by colinsurprenant** [comment]
@flaper87 the newer version of the library simply exposes a timeout option which defaults to 5s and passes that option to the udp and tcp sockets instead of relying on the defaults for theses. We can look into also exposing that timeout as a plugin option if needed.

**Comment by colinsurprenant** [comment]
v3.0.7 published.

**Pull request merged**